src/server/types.ts

```diff
@@ -189,9 +189,30 @@ export interface FreshContext<
   destination: router.DestinationKind;
   params: Record<string, string>;
   isPartial: boolean;
+  /**
+   * An object where data can be assigned to and that is accessible
+   * across middlewares, handlers and routes.
+   * @example
+   * ```tsx
+   * // routes/_middleware.ts
+   * export type State = { data: string };
+   *
+   * export async function handler(_req: Request, ctx: FreshContext<State>) {
+   *   ctx.state.data = "myData";
+   *   return ctx.next();
+   * }
+   * // routes/data.tsx
+   * export const handler: Handler<void, State> = (_req, ctx) => {
+   *   return new Response(`middleware data: ${ctx.state.data}`);
+   * };
+   * ```
```

Comment:
Not sure if we need this large example here. It was adapted from https://fresh.deno.dev/docs/concepts/middleware. Maybe we should link to that page instead via `@link`? It feels a bit odd though to link there, since the page is named "Middleware" (though it explains about `state` too).